services/nycrc/nycrc.service.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { coveragePercentage } = require('../color-formatters')
+const {
+  ConditionalGithubAuthV3Service,
+} = require('../github/github-auth-service')
+const { fetchJsonFromRepo } = require('../github/github-common-fetch')
+const { InvalidResponse, NotFound } = require('..')
+
+const nycrcSchema = Joi.object({
+  branches: Joi.number()
+    .min(0)
+    .max(100),
+  lines: Joi.number()
+    .min(0)
+    .max(100),
+}).required()
+
+const pkgJSONSchema = Joi.object({
+  c8: Joi.object({
+    branches: Joi.number()
+      .min(0)
+      .max(100),
+    lines: Joi.number()
+      .min(0)
+      .max(100),
+  }).optional(),
+  nyc: Joi.object({
+    branches: Joi.number()
+      .min(0)
+      .max(100),
+    lines: Joi.number()
+      .min(0)
+      .max(100),
+  }).optional(),
+}).required()
+
+const documentation = `<p>
+  Create a code coverage badge, based on thresholds stored in a
+  <a href="https://github.com/istanbuljs/nyc#common-configuration-options">.nycrc config file</a>
+  on GitHub.
+</p>`
+
+module.exports = class Nycrc extends ConditionalGithubAuthV3Service {
+  static get category() {
+    return 'coverage'
+  }
+
+  static get route() {
+    return {
+      base: 'nycrc',
+      // TODO: eventually add support for .yml and .yaml:
+      pattern: ':user/:repo',
+      queryParamSchema: Joi.object({
+        config: Joi.string()
+          .valid('.nycrc', '.nycrc.json', 'package.json')
```

Comment:
It's common with other badges to allow users to specify the repo path to the file, since they're not always in the root of the repo. I suspect we'll need to do something similar here, and could use two different query params (one for directory path and one for file name) if we want to have validation on the specific file names that'll be accepted

Reply:
perhaps we should just default to .nycrc, and allow an arbitrary path?